addons/controllers/suite_test.go

```diff
@@ -275,6 +283,39 @@ var _ = BeforeSuite(func(done Done) {
 	err = testutil.CreateResources(f, cfg, dynamicClient)
 	Expect(err).ToNot(HaveOccurred())
 
+	labelMatch, _ := labels.NewRequirement("webhook-cert", selection.Equals, []string{"self-managed"})
```

Comment:
suggestion: is it possible we put the following webhook setup into a separate function ? `BeforeSuite` holds 200+ lines which makes the maintenance hard

Reply:
Agreed, we should make the webhook setup related code (starting from line 280) as a separate function

Reply:
+1 a separate helper function for webhook setup

Reply:
Fixed. Added a helper function and refactored all the affected tests